plugins/modules/ndo_virtual_port_channel_interface.py

```diff
@@ -0,0 +1,383 @@
+            [KVPair("uuid", virtual_port_channel_interface_uuid) if virtual_port_channel_interface_uuid else KVPair("name", virtual_port_channel_interface)],
+        )
+        if match:
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+    else:
+        mso.existing = mso.previous = existing_virtual_port_channel_interfaces
+
+    if state == "present":
+
+        if interface_policy_group and not interface_policy_group_uuid:
+            fabric_policy_template = MSOTemplate(mso, "fabric_policy", interface_policy_group.get("template"))
+            fabric_policy_template.validate_template("fabricPolicy")
+            interface_policy_group_uuid = fabric_policy_template.get_interface_policy_group_uuid(interface_policy_group.get("name"))
+
+        if match:
+            if virtual_port_channel_interface and match.details.get("name") != virtual_port_channel_interface:
+                ops.append(dict(op="replace", path="{0}/{1}/name".format(path, match.index), value=virtual_port_channel_interface))
+                match.details["name"] = virtual_port_channel_interface
+
+            if description and match.details.get("description") != description:
+                ops.append(dict(op="replace", path="{0}/{1}/description".format(path, match.index), value=description))
+                match.details["description"] = description
+
+            if node_1 and match.details.get("node1Details", {}).get("node") != node_1:
+                ops.append(dict(op="replace", path="{0}/{1}/node1Details/node".format(path, match.index), value=node_1))
+                match.details["node1Details"]["node"] = node_1
+
+            if node_2 and match.details.get("node2Details", {}).get("node") != node_2:
+                ops.append(dict(op="replace", path="{0}/{1}/node2Details/node".format(path, match.index), value=node_2))
+                match.details["node2Details"]["node"] = node_2
+
+            if interface_policy_group_uuid and match.details.get("policy") != interface_policy_group_uuid:
+                ops.append(dict(op="replace", path="{0}/{1}/policy".format(path, match.index), value=interface_policy_group_uuid))
+                match.details["policy"] = interface_policy_group_uuid
+
+            if interfaces_node_1 and interfaces_node_1 != match.details.get("node1Details", {}).get("memberInterfaces"):
+                ops.append(dict(op="replace", path="{0}/{1}/node1Details/memberInterfaces".format(path, match.index), value=interfaces_node_1))
+                match.details["node1Details"]["memberInterfaces"] = interfaces_node_1
+
+            if interfaces_node_2 and interfaces_node_2 != match.details.get("node2Details", {}).get("memberInterfaces"):
+                ops.append(dict(op="replace", path="{0}/{1}/node2Details/memberInterfaces".format(path, match.index), value=interfaces_node_2))
+                match.details["node2Details"]["memberInterfaces"] = interfaces_node_2
+
+            if interface_descriptions:
```

Comment:
just confirmation can we set descriptions to node that are not part of the VPV or should we validate this? or does ndo error in this scenario?

Reply:
Same thing, The API's errors out if they are matching what are the current nodes/interfaces config